.travis.yml

```diff
@@ -1,63 +1,93 @@
+  - if [ "$TRAVIS_OS_NAME" == "linux" ]; then travis_retry sudo apt-get update -q; fi
+  - if [ "$TRAVIS_OS_NAME" == "linux" ]; then travis_retry sudo apt-get install -y libcairo2-dev; fi
+  - if [ "$TRAVIS_OS_NAME" == "osx" ]; then brew update; fi
+  - if [ "$TRAVIS_OS_NAME" == "osx" ]; then brew install pkg-config || true; fi
+  - if [ "$TRAVIS_OS_NAME" == "osx" ]; then brew install cairo || true; fi
+  - if [ "$TRAVIS_OS_NAME" == "osx" ]; then export PKG_CONFIG_PATH="/usr/local/opt/libffi/lib/pkgconfig"; fi
+  - if [ "$TRAVIS_OS_NAME" == "osx" ]; then brew unlink python@2 || true; fi
+  - if [ "$TRAVIS_OS_NAME" == "osx" ]; then brew install python || true; fi
+  - if [ "$TRAVIS_OS_NAME" == "osx" ]; then python3 -m pip install virtualenv; fi
+  - if [ "$TRAVIS_OS_NAME" == "osx" ]; then virtualenv ../venv -p python3; fi
+  - if [ "$TRAVIS_OS_NAME" == "osx" ]; then source ../venv/bin/activate; fi
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then python -m pip install --upgrade setuptools; fi
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then python -m pip install --upgrade pytest flake8 "sphinx<3" sphinx_rtd_theme coverage codecov hypothesis attrs; fi
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then python -m pip install --upgrade mypy || true; fi
 script:
-  - python -m coverage run --branch setup.py test
-  - python -m codecov --required || true
-  - python -m flake8 .
-  - python setup.py sdist
-  - python setup.py bdist
-  - python setup.py install --root=_root
-  - python setup.py install --root="$(pwd)"/_root_abs
-  - python setup.py bdist_egg
-  - python setup.py bdist_wheel
-  - python setup.py install --root=_root_setup
-  - if [[ "${TRAVIS_PYTHON_VERSION:0:4}" != "pypy" ]] ; then python -m pip install .; fi
-  - python -m sphinx -W -a -E -b html -n docs docs/_build
+  #windows build
+  - if [ "$TRAVIS_OS_NAME" == "windows" ]; then source .travis/runPycairox64.sh; fi
+  - if [ "$TRAVIS_OS_NAME" == "windows" ]; then source .travis/runPycairox86.sh; fi
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then python setup.py sdist; fi
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then python setup.py bdist; fi
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then python setup.py install --root=_root; fi
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then python setup.py install --root="$(pwd)"/_root_abs; fi
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then python setup.py bdist_wheel; fi
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then python setup.py install --root=_root_setup; fi
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then if [[ "${TRAVIS_PYTHON_VERSION:0:4}" != "pypy" ]] ; then python -m pip install .; fi; fi
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then python -m sphinx -W -a -E -b html -n docs docs/_build; fi
+
+cache:
+  directories:
+    - $HOME/.cache/pip
+    - $HOME/Library/Caches/Homebrew
+
+before_cache:
+  - brew cleanup
+  - rm -f $HOME/.cache/pip/log/debug.log
+branches:
```

Comment:
this seems to disable CI for all other branches. is that on purpose?

Reply:
It's for speeding up builds by caching and doesn't stop for all other branches(I mean OS). I have done a [test build](https://travis-ci.org/github/naveen521kk/pycairo/builds/712093851) and it didn't cause any issues.

Reply:
In case it wasn't clear: Travis-CI for this PR doesn't run, but it should.

Reply:
I just misunderstood, I was seeing `before_cache` and was responding very sorry.
 Yeah, I did it on purpose, and if needed, I will enable it to build for all the PR also. The release I thought it would be good for releases to be built. If I am wrong I am changing it.

Reply:
Should I also add `master` in that list so that Travis builds build all the PR's made to master? @lazka

Reply:
tests should run on all branches.

Reply:
Ok, then I will remove the branch specific changes.